packages/FileUpload/src/index.tsx

```diff
@@ -1,80 +1,97 @@
 import Button from "@blaze-react/button";
-import React, { useEffect, useRef } from "react";
+import React, { useEffect, useRef, useState } from "react";
+import FileList from "./FileList";
+
 interface IFileUploadProps {
-  handleDrop: (...args: any[]) => any;
   children?: any;
+  onChange: (...args: any[]) => void;
 }
 const FileUpload: React.SFC<IFileUploadProps> = ({
   children,
-  handleDrop: handleDropProp,
+  onChange,
   ...attr
 }) => {
+  const [previewImages, setPreviewImages]: any[] = useState([]);
+  const [filesToUpload, setFilesToUpload]: any[] = useState([]);
   const area: any = useRef(null);
   const selectFile: any = useRef(null);
   const handleDragover = (event: any): void => {
     event.stopPropagation();
     event.preventDefault();
   };
+
+  useEffect(() => {
+    const handleDrop = (event: any) => {
+      event.preventDefault();
+      event.stopPropagation();
+      let { dataTransfer: { files = {} } = {} } = event;
+      files = Object.values(files);
+      processFiles(files);
+    };
+    const { current: currentArea } = area;
+    currentArea.addEventListener("dragover", handleDragover);
+    currentArea.addEventListener("drop", handleDrop);
```

Comment:
Here we don't need to remove the listeners in any case?

Reply:
Good question, let me check it.